tests/protocols/test_websocket.py

```diff
@@ -731,9 +743,12 @@ async def websocket_session(uri):
             websocket_session(f"ws://127.0.0.1:{unused_tcp_port}")
         )
         await asyncio.sleep(0.1)
-        task.cancel()
         send_accept_task.set()
 
+    task.cancel()
+    assert response is not None
+    assert response.status_code == 500, response.text
+    assert response.text == "Internal Server Error"
     assert disconnect_message == {"type": "websocket.disconnect", "code": 1006}
```

Comment:
Hmm... On a second thought... This doesn't make sense. Why are we even sending a `websocket.disconnect` when the handshake was not even completed? :thinking: 

I think this was on purpose, but thinking about it again, does it make sense?

Reply:
I need to check the issues/PRs about this decision. It shouldn't be a blocker for this PR tho.

Reply:
According to https://github.com/django/asgiref/issues/364 it's fine.